lib/tasks/svg.rake

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+namespace :svg do
+  desc "rerender SVG for molecules  \n
+     args[:all] default to false, if true, all molecules will be rerendered, \n
+     if false, only molecules with missing SVG will be rerendered"
+  task :molecule, %i[all from to] => [:environment] do |_t, args|
+    #   if true, render all molecules, otherwise only render those that have missing svg
+    # args.with_ketcher is a boolean: default to true, always check ketcher rendering service is available
+    all = args[:all] == 'true'
+    from = args[:from].presence&.to_i || 0
+    to = args[:to].presence&.to_i || 0
+    molecule_scope = Molecule.all
+    molecule_scope = molecule_scope.where(id: from..) if from.positive?
+    molecule_scope = molecule_scope.where(id: ..to) if to.positive?
+    message_header = 'Reprocessing '
+    message_header += all ? 'all' : 'missing'
+    message_header += ' molecule SVGs: '
+    puts "#{message_header}Ketcher rendering service seems to #{ketcher_running ? '' : 'not'} be available"
+
+    # if ketcher_only && !ketcher_running
+    #   puts "#{message_header}Ketcher rendering service not available, exiting"
+    #   break
+    # end
```

Comment:
The `ketcher_only` variable is also nonexistent in the rest of the code. Clarification on its usage & existence needed.

Either:
1) this conditional should be implemented, to avoid the task running to completion with the ketcher service unavailable.
2) the user should be made aware in line 19 (& 67), that they should exit the task if the ketcher service is unavailable.

Reply:
fx and moved to https://github.com/ComPlat/chemotion_ELN/pull/1308